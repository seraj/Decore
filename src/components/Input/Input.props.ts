import ThemeProps from "../../theme/Theme.props";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export default interface InputProps
  extends ThemeProps,
    React.InputHTMLAttributes<HTMLInputElement> {
  fluid?: boolean;
}
