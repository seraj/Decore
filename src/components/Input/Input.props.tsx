export default interface InputProps {
    defaultValue?: string;
    value?: string;
    /** Additional classes */
    className?: string;
    /** Renders a disabled button and prevents future behavior */
    disabled?: boolean;
    /** Will render a simple input without styles */
    simple?: boolean;
    /** Change radius of input */
    radius?: number;
    /** Callback function to change the value */
    onChange?: () => void,
    /** set type of input */
    type?: string;
}

