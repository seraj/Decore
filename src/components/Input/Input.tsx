import * as React from "react";
import StyledInput from "./styled/StyledInput";

interface InputProps {
    defaultValue?: string;
    value?: string;
    /** Additional classes */
    className?: string;
    /** Renders a disabled button and prevents future behavior */
    disabled?: boolean;
    /** Will render a simple input without styles */
    simple?: boolean;
    /** Change radius of input */
    radius?: number;
    /** Callback function to change the value */
    onChange?: () => void
}

const Input: React.FC<InputProps> = ({
    className,
    disabled = false,
    simple = false,
    radius = 0,
    defaultValue = '',
    value = '',
    onChange
}) => (
    <StyledInput
        defaultValue={defaultValue}
        value={value}
        className={className}
        disabled={disabled}
        simple={simple}
        radius={radius}
        onChange={onChange} />
);

export default Input;
