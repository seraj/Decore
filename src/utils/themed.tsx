import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
