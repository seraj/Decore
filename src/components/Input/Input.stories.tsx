import { text, withKnobs, boolean, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Input from "./Input";

const stories = storiesOf("Components/Input", module);
stories.addDecorator(withKnobs);

stories.add(
    "default",
    wInfo()(() => (
        <Input
            type={text("type", "text")}
            onChange={action("onChange")}
            defaultValue={text("defaultValue", "")}
            value={text("defaultValue", "")}
            className={text("className", "")}
            disabled={boolean("disabled", false)}
            simple={boolean("simple", false)}
            radius={number("border-radius", 0)}
        />
    ))
);
