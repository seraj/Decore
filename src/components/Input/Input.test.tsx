import * as React from "react";

import themed from "../../utils/themed";
import Input from "./Input";

describe("Input", () => {
  it("should render correctly", () => {
    const component = themed(
      <Input>
        Test
      </Input>,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
