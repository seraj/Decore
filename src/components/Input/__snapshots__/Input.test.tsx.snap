// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
.c0 {
  border: 1.5px solid #999999;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  background-color: #ffffff;
  padding: 1rem 0.8rem;
  height: 42px;
  border-radius: 3px;
  color: #666666;
}

.c0:hover {
  border-color: #333333;
  color: #666666;
}

.c0:active {
  color: #333333;
}

.c0:focus {
  color: #333333;
  border-color: #ffffff;
}

.c0:disabled {
  border: none;
  background: #E5E5E5;
  color: #999999;
  cursor: not-allowed !important;
  pointer-events: none;
}

<input
  className="c0"
/>
`;
